### ISSUE TITLE
Using ItemStack#withAmount of 127 or higher results in a smaller or ghost ItemStack

### ISSUE THREAD

**Issue by AlexDicy** [Minestom/Minestom#425]
Using this code to give an ItemStack of 128 items results in a ghost stack
```java
player.getInventory().addItemStack(ItemStack.of(Material.GRASS_BLOCK, 128));
```
![image](https://user-images.githubusercontent.com/11839341/130335096-871883e3-6aed-4061-8304-12ad8e63da6e.png)

Using this code to give an ItemStack of 127 items twice results in a single stack of 64, 2 ghost stacks of 64 and a ghost stack of 62
```java
player.getInventory().addItemStack(ItemStack.of(Material.GRASS_BLOCK, 127));
player.getInventory().addItemStack(ItemStack.of(Material.GRASS_BLOCK, 127));
```
![image](https://user-images.githubusercontent.com/11839341/130335197-432b44d7-1cda-4f55-8831-b401e6b88ed4.png)
when moved around
![image](https://user-images.githubusercontent.com/11839341/130335211-d6f58f56-7915-45d8-a271-209efcadc9ef.png)

**Comment by Eoghanmc22** [comment]
i think the first case is happening because 128 as a byte is -128 and that is less than 0 and there might be some code that doesn't like that in minestom. im like 90% sure this is caused by the default stacking rule but might have multiple causes
the 2nd case is a bug where the "ADD" TransactionType checks "itemAmount == maxSize" on line 34 where it should be "itemAmount >= maxSize"

**Comment by TheMode** [comment]
Fixed by 87819ba80653e8be28f8087640a46b1fffb53168
